compiler/qsc_fir/src/fir.rs

```diff
@@ -765,6 +759,56 @@ impl Display for CallableDecl {
         write!(indent, "\ninput: {}", self.input)?;
         write!(indent, "\noutput: {}", self.output)?;
         write!(indent, "\nfunctors: {}", self.functors)?;
+        write!(indent, "\nimplementation: {}", self.implementation)?;
+        Ok(())
+    }
+}
+
+/// A callable implementations.
+#[derive(Clone, Debug, PartialEq)]
+pub enum CallableImplementation {
+    /// An intrinsic callable implementation.
+    Intrinsic(NodeId, Span),
+    /// A specialized callable implementation.
+    Specialized(SpecializedImplementation),
+}
+
+impl Display for CallableImplementation {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        let mut indent = set_indentation(indented(f), 0);
+        match self {
+            CallableImplementation::Intrinsic(node_id, span) => {
+                write!(indent, "Instrinsic {node_id} {span}")?;
+            }
+            CallableImplementation::Specialized(spec_impl) => {
+                write!(indent, "Specialized:")?;
+                indent = set_indentation(indent, 1);
+                write!(indent, "\n{spec_impl}")?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+/// A specialized implementation.
+#[derive(Clone, Debug, PartialEq)]
+pub struct SpecializedImplementation {
```

Comment:
nitpick: same here, we usually abbreviate to "Spec" so something like `SpecImpl` would be shorter and match other patterns.

Reply:
Renamed.